src/pitaya_etcd_nats_cluster/src/rpc_server.rs

```diff
@@ -13,9 +15,24 @@ use tokio::sync::{mpsc, oneshot, RwLock};
 const RPC_LATENCY_METRIC: &str = "rpc_latency";
 const RPCS_IN_FLIGHT_METRIC: &str = "rpcs_in_flight";
 
+struct RpcServerState {
+    connection: asynk::Connection,
+    close_sender: oneshot::Sender<()>,
+}
+
+impl RpcServerState {
+    async fn close(self) -> Result<(), Error> {
+        // TODO(victor-carvalho) we have to wait on connection close
+        let _ = self.close_sender.send(());
+        self.connection.close().await.map_err(Error::Nats)
+    }
+}
+
+type NatsRpcServerState = Arc<RwLock<Option<RpcServerState>>>;
```

Comment:
Since the type is private, you can remove the Nats prefix.

Reply:
I only added the Nats because RpcServerState already exists